pkg/migrate/migrate.go

```diff
@@ -39,50 +39,52 @@ func Cli() {
 	flag.BoolVar(&verboseCopy, "verbose-copy", false, "show output from the rsync command used to copy data between PVCs")
 
 	flag.Parse()
-	Migrate(sourceSCName, destSCName, rsyncImage, setDefaults, verboseCopy)
-}
 
-// Migrate moves data and PVCs from one StorageClass to another
-func Migrate(sourceSCName, destSCName, rsyncImage string, setDefaults, verboseCopy bool) {
 	// setup k8s
 	cfg, err := config.GetConfig()
 	if err != nil {
-		fmt.Printf("failed to get config: %v\n", err)
+		fmt.Printf("failed to get config: %s\n", err.Error())
 		os.Exit(1)
 	}
 
 	clientset, err := k8sclient.NewForConfig(cfg)
 	if err != nil {
-		fmt.Printf("failed to create kubernetes clientset: %v\n", err)
+		fmt.Printf("failed to create kubernetes clientset: %s\n", err.Error())
 		os.Exit(1)
 	}
 
-	err = validateStorageClasses(os.Stdout, clientset, sourceSCName, destSCName)
+	err = Migrate(context.TODO(), os.Stdout, clientset, sourceSCName, destSCName, rsyncImage, setDefaults, verboseCopy)
 	if err != nil {
 		fmt.Printf("%s\n", err.Error())
 		os.Exit(1)
 	}
+}
 
-	matchingPVCs, newPVCs, originalRetentionPolicies, originalPVNames, namespaces, err := getPVCs(context.TODO(), os.Stdout, clientset, sourceSCName, destSCName)
+// Migrate moves data and PVCs from one StorageClass to another
+func Migrate(ctx context.Context, w io.Writer, clientset k8sclient.Interface, sourceSCName, destSCName, rsyncImage string, setDefaults, verboseCopy bool) error {
+	err := validateStorageClasses(ctx, w, clientset, sourceSCName, destSCName)
 	if err != nil {
-		fmt.Printf("%s\n", err.Error())
-		os.Exit(1)
+		return fmt.Errorf("%w", err)
 	}
 
-	err = scaleDownPods(os.Stdout, clientset, matchingPVCs)
+	matchingPVCs, newPVCs, originalRetentionPolicies, originalPVNames, namespaces, err := getPVCs(ctx, w, clientset, sourceSCName, destSCName)
 	if err != nil {
-		fmt.Printf("failed to scale down pods: %v\n", err)
-		os.Exit(1)
+		return fmt.Errorf("%s", err)
+	}
+
+	err = scaleDownPods(ctx, w, clientset, matchingPVCs)
+	if err != nil {
+		return fmt.Errorf("failed to scale down pods: %w", err)
 	}
 
 	// create a pod for each PVC migration, and wait for it to finish
-	fmt.Printf("\nCopying data from %s PVCs to %s PVCs\n", sourceSCName, destSCName)
+	_, _ = fmt.Fprintf(w, "\nCopying data from %s PVCs to %s PVCs\n", sourceSCName, destSCName)
```

Comment:
id probably remove the underscores for the print statements. i get the point but it doesnt read well.

Reply:
every single one shows up in my gutter 😆 

Honestly I wish there was a good way to handle these, but what even is there to handle?